source/json.c

```diff
@@ -180,8 +180,13 @@ int aws_json_const_iterate_object(
 
     const cJSON *key = NULL;
     cJSON_ArrayForEach(key, cjson) {
+        bool should_continue = true;
         struct aws_byte_cursor key_cur = aws_byte_cursor_from_c_str(key->string);
-        if (!on_member(&key_cur, (struct aws_json_value *)key, user_data)) {
+        if (on_member(&key_cur, (struct aws_json_value *)key, &should_continue, user_data)) {
+            return AWS_OP_ERR;
```

Comment:
trivial/consistency: other parts of the function do `goto done;` when an error happens

be uniform. Either always `goto done;` on error, or always return and get rid of the gotos